### DISCUSSION TITLE
Kate syntax xml for Pandoc

### DISCUSSION THREAD

**Discussion by JLPLabs** [teal-language/tl#641] [Show and tell]
I'm working on a blog series covering my experiences with Teal and Cyan.

To keep it simple I'm using these tools:
* ReStructured Text files (.rst)
* Pandoc
* Bash script to drive static web generation

To enable syntax-colored code blocks I needed to create tools that Pandoc recognizes:
* Kate style xml syntax parsing
* Pandoc style "theme" for syntax coloring

MY GOAL was an **extremely** sparse markup. You will see that the only coloring is:
* _Declaration_ of variables.
* _Declaration_ of fields/methods.
* Basic strings  (as opposed to long string format, which is not handled)
* Comments

So, attached are:
1. teal.xml
2. mysparselight.theme
3. mysparsedark.theme
4. example light screenshot
5. example dark screenshot

FUN TO NOTE: 'record' and 'enum' are valid identifiers in Teal. Most (all?) of the editors I looked at that can handle Teal syntax coloring CAN NOT handle 'record' or 'enum' as identifiers. And they kind of lose their minds for a little bit, before recovering.

My parser **can** handle these. See lines 11 and 14, below, for example.

[xml_and_themes.zip](https://github.com/teal-language/tl/files/11017735/xml_and_themes.zip)

<img width="631" alt="example_light" src="https://user-images.githubusercontent.com/4744024/226338948-412aeaeb-ead4-4fdb-890b-3281dfe577bc.png">

<img width="631" alt="example_dark" src="https://user-images.githubusercontent.com/4744024/226338980-4a7e7650-2971-444c-ab91-6c570c8d176e.png">

**Comment by bjornbm** [comment]
Don't forget to let us know when you publish you blog series!

**Reply by JLPLabs** [reply]
You can find the first two entries at: www.jlplabs.com
